test/utils/Permit2Utils.sol

```diff
@@ -0,0 +1,73 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.23;
+
+import {ISignatureTransfer} from "../interfaces/ISignatureTransfer.sol";
+import {Vm} from "forge-std/Vm.sol";
+import {IPermit2} from "../interfaces/IPermit2.sol";
+
+library Permit2Utils {
```

Comment:
Nice! Would love to have this work well with our existing testing utils like `TestKeyManager`, etc. since we have different ways of signing depending on the keyType & if it is the root key or an external signer. For example, external signers will have to sign over 7739 wrapped types. I can add it in if you're ok with it!

Reply:
Updated! LMK if that's what you had in mind.